src/cli/serve/serve.js

```diff
@@ -322,18 +312,13 @@ export default function (program) {
     const configs = [getConfigPath(), ...getEnvConfigs(), ...(opts.config || [])];
     const serverlessMode = getServerlessProjectMode(opts);
 
-    // we "unshift" .serverless. config so that it only overrides defaults
-    if (serverlessMode) {
-      maybeAddConfig(`serverless.yml`, configs, 'push');
```

Comment:
I'm intentionally not adding `serverless.yml` because it's added later on when `bootstrap.ts` finds out in the config that a specific `serverless` mode is required.